test/operator/discr_ops_test.py

```diff
@@ -0,0 +1,242 @@
+# Copyright 2014, 2015 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+
+# External module imports
+import numpy as np
+import pytest
+
+# ODL imports
+from odl.discr.lp_discr import uniform_discr, FunctionSpace, IntervalProd
+from odl.operator.discr_ops import (finite_diff, DiscretePartDeriv,
+                                    DiscreteGradient, DiscreteDivergence)
+from odl.space.ntuples import Rn
+from odl.set.domain import Rectangle
+
+
+def test_finite_diff():
+    """Finite differences test. """
+
+    ndim = 10
+    f = np.arange(ndim, dtype=float)
```

Comment:
I.m.o using functions such as these does not help, simply doing a very simple `f = [0, 1, 2, 3]` should do splendidly.